src/sentry/auth/access.py

```diff
@@ -410,17 +645,95 @@ def has_project_membership(self, project: Project) -> bool:
         return self.has_project_access(project)
 
 
+class ApiOrganizationGlobalMembership(ApiOrganizationGlobalAccess):
+    """Access to all an organization's teams and projects with simulated membership."""
+
+    @property
+    def team_ids_with_membership(self) -> FrozenSet[int]:
+        return self.accessible_team_ids
+
+    @property
+    def project_ids_with_team_membership(self) -> FrozenSet[int]:
+        return self.accessible_project_ids
+
+    def has_team_membership(self, team: Team) -> bool:
+        return self.has_team_access(team)
+
+    def has_project_membership(self, project: Project) -> bool:
+        return self.has_project_access(project)
+
+
+@dataclass
 class OrganizationlessAccess(Access):
```

Comment:
Interestingly, it seemed really simple to implement the entire OrganizationlessAccess class without any db access, just the ApiAuthState object, which I was already sharing implementation with the existing one (it encapsulates `permissions` and `_sso_state`.  This allows me to share this access implementation across both the DB and service Access versions.